core/src/main/java/jenkins/util/JSONSignatureValidator.java

```diff
@@ -321,5 +323,25 @@ protected Set<TrustAnchor> loadTrustAnchors(CertificateFactory cf) throws IOExce
         return anchors;
     }
 
+    /**
+     * Extracts a user-friendly message from an exception chain.
+     *
+     * @param e the exception to extract the message from
+     * @return a concise, readable error message
+     */
+    private String getRootCauseMessage(Throwable e) {
```

Comment:
Optional suggestion to annotate the argument so that spotbugs knows that the implementation expects that the argument is not null.  A null pointer exception will be reported if a null value is passed as the argument.
```suggestion
    private String getRootCauseMessage(@NonNull Throwable e) {
```